omp/matrix/csr_kernels.cpp

```diff
@@ -1020,6 +1020,126 @@ GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_CSR_ADD_SCALED_IDENTITY_KERNEL);
 
 
+bool csr_lookup_allowed(matrix::sparsity_type allowed,
+                        matrix::sparsity_type type)
+{
+    return ((static_cast<int>(allowed) & static_cast<int>(type)) != 0);
+}
+
+
+template <typename IndexType>
+bool csr_lookup_try_full(IndexType row_len, IndexType col_range,
+                         matrix::sparsity_type allowed, int64& row_desc)
+{
+    using matrix::sparsity_type;
+    bool is_allowed = csr_lookup_allowed(allowed, sparsity_type::full);
+    if (is_allowed && row_len == col_range) {
+        row_desc = static_cast<int>(sparsity_type::full);
+        return true;
+    }
+    return false;
+}
+
+
+constexpr static int csr_lookup_block_size = 32;
```

Comment:
Isn't that `device_sparsity_lookup::block_size` (or however you might rename that)?